app/assets/scripts/components/upload-form.js

```diff
@@ -62,9 +62,8 @@ var UploadForm = React.createClass({
           failures = ['No data provided'];
           this.logOutput(failures);
         }
-        // Check header on line 1
+        // Check header on first line
         if (line === 0) this.checkHeader(data, failures);
```

Comment:
If I understand correctly, I think this has been fixed. The logOutput function, which is now called setErrorState, now only receives the array of logs and sets the state in response. The metadata and failures are only appended to in the main csvreader stream, and aren't modified by any other functions until they are rendered.